compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -67,7 +68,10 @@ trait TypeOps { this: Context => // TODO: Make standalone object.
           if (thiscls.derivesFrom(cls) && pre.baseTypeRef(thiscls).exists) {
             if (theMap != null && theMap.currentVariance <= 0 && !isLegalPrefix(pre)) {
               ctx.base.unsafeNonvariant = ctx.runId
-              AnnotatedType(pre, Annotation(defn.UnsafeNonvariantAnnot, Nil))
+              pre match {
+                case AnnotatedType(_, ann) if ann.symbol == defn.UnsafeNonvariantAnnot => pre
```

Comment:
No it's related. Before we were running into the member/asSeenFrom mutual recursion we stackoverflowed here. Avoiding this by not piling up the same annotation makes sense independently.